src/neuroglancer/datasource/computed/tensorflow/frontend.ts

```diff
@@ -0,0 +1,251 @@
+import {InferenceRequest, InferenceResult, TENSORFLOW_COMPUTATION_RPC_ID, TENSORFLOW_INFERENCE_RPC_ID, TensorflowArray, TensorflowComputationParameters} from 'neuroglancer/datasource/computed/tensorflow/base';
+import {DataType, VolumeType} from 'neuroglancer/sliceview/volume/base';
+import {CANCELED, CancellationToken} from 'neuroglancer/util/cancellation';
+import {verifyFloat, verifyString} from 'neuroglancer/util/json';
+import {registerPromiseRPC, registerSharedObjectOwner, RPCPromise} from 'neuroglancer/worker_rpc';
+
+/**
+ * Dict-style object that represents a pending inference request.
+ */
+interface PendingInference {
+  // TF.js-compatible typed array, to be used as input.
+  array: TensorflowArray;
+  // If cancelled, reject(CANCELED) is called instead of performing inference.
+  cancellationToken: CancellationToken;
+  // The priority of the output chunk. Requests are handled in priority order.
+  priority: number;
+  // Resolve/Reject lambdas for the Promise awaiting inference.
+  reject: (message: any) => void;
+  resolve: (inferenceResult: InferenceResult) => void;
+}
+
+@registerSharedObjectOwner(TENSORFLOW_COMPUTATION_RPC_ID)
+export class TensorflowComputation extends VolumeComputationFrontend {
+  params: TensorflowComputationParameters;
+
+  // The queue of pending inference requests. Maintained in order of increasing
+  // priority, so that pop() always returns the next appropriate request.
+  private inferenceQueue_: PendingInference[] = [];
+  // True iff the inference loop is in operation.
+  private running_ = false;
+
+  constructor(params: TensorflowComputationParameters, private model_: InferenceModel) {
+    super(params);
+  }
+
+  /**
+   * Adds a TF.js-compatible typed array to the inference queue and starts the
+   * inference loop, if it isn't already running. Returns a promise that
+   * resolves with the prediction output.
+   *
+   * Creates a PendingInference object and inserts it into the correct position
+   * in the inference queue, which is guaranteed to be less than the length of
+   * the backend computation queue in length. As such, it is most likely
+   * faster to simply iterate the queue to insert the request rather than using
+   * a more sophisticated method like binary insertion or a tree-based queue.
+   * @param array the input data to run inference over
+   * @param priority
+   * @param cancellationToken
+   */
+  predict(array: TensorflowArray, priority: number, cancellationToken: CancellationToken):
+      RPCPromise<InferenceResult> {
+    return new Promise((resolve, reject) => {
+             let i;
+             const queue = this.inferenceQueue_;
+             for (i = 0; i < queue.length && queue[i].priority < priority; ++i) {
+             }
+             queue.splice(i, 0, {array, cancellationToken, priority, resolve, reject});
+             this.startInference();
+           })
+        .then((result: InferenceResult) => {
+          return {value: result};
```

Comment:
To avoid copying the data, use:

{value: result, transfers: [result.outputBuffer.buffer]}

assuming that you don't need to hang on to the outputBuffer.

Reply:
Done.